src/components/spinner/Spinner.tsx

```diff
@@ -77,33 +80,42 @@ export const Spinner = forwardRef<HTMLSpanElement, Props>((props, ref) => {
         </SpinnerInner>
       </SpinnerWrapper>
       {!isLoading &&
-        (children ?? (hasPlaceholder && <Placeholder className={className} />) ?? null)}
+        (children ?? (hasPlaceholder && <SpinnerPlaceholder className={className} />) ?? null)}
     </>
   );
 });
 
 Spinner.displayName = 'Spinner';
 
-const SpinnerWrapper = styled.span<{ inline: boolean }>`
+const SpinnerWrapper = styled.span<{ inline: boolean; isLoading?: boolean }>`
   display: ${(props) => (props.inline ? 'inline-block' : 'block')};
   position: relative;
-  ${(props) => (props.inline ? 'margin: 0 var(--echoes-dimension-space-50);' : '')}
+  ${(props) =>
+    props.inline && props.isLoading ? 'margin: 0 var(--echoes-dimension-space-50);' : ''}
+
+  &.echoes-button-spinner-wrapper {
```

Comment:
I'm not super happy with this pattern, I wanted to use `${ButtonStyled} & {}` instead but ran into some issues that I didn't understand so I tried to move faster by going around the issue.

Reply:
Yeah, not a fan either. How about:

> Button.tsx
```
const StyledSpinner = styled(Spinner)`
  margin-right: var(--echoes-dimension-size-75);

  background:
    linear-gradient(0deg, var(--button-spinner-color) 50%, transparent 50% 100%) border-box,
    linear-gradient(90deg, var(--button-spinner-color) 25%, transparent 75% 100%) border-box;
`;
```

Reply:
It would work for this part of the style but not for overriding the margin of the wrapper to 0. Since the classname prop is propagated to the inner spinner.

Another aspect I don't like about this, is that we are moving some style that impact the spinner component in another place unrelated to the spinner itself. I think when looking at the spinner style we should be able to also see all stuff that could impact it's style.